quic-client/src/lib.rs

```diff
@@ -140,9 +142,15 @@ impl QuicConfig {
     pub fn update_client_certificate(
         &mut self,
         keypair: &Keypair,
-        ipaddr: IpAddr,
+        ipaddr: Option<IpAddr>,
     ) -> Result<(), RcgenError> {
-        let (cert, priv_key) = new_self_signed_tls_certificate(keypair, ipaddr)?;
+        let (cert, priv_key) = if let Some(ip) = ipaddr {
+            let res = new_self_signed_tls_certificate(keypair, ip)?;
+            self.ip = ip;
+            res
+        } else {
+            new_self_signed_tls_certificate(keypair, self.ip)?
+        };
```

Comment:
I found it more convenient to just store the IP in QuicConfig since that's where we the socket anyway, and bubbling it up just to store it somewhere else seems like a little too much added complexity.